lib/actions/field-trip.js

```diff
@@ -249,17 +406,355 @@ function planInbound (request) {
+          fieldTripModuleConfig,
+          leg.mode
+        )
+        let remainingTripCapacity = legModeCapacity - capacityInUse
+        if (remainingTripCapacity < minimumAllowableRemainingCapacity) {
+          // This trip is already too "full" to allow any addition field trips
+          // on board. Ban this trip in future searches and don't use this
+          // itinerary in final results (set trip and itinerary capacity to 0).
+          remainingTripCapacity = 0
+        }
+
+        // always ban trips found in itineraries so that subsequent searches
+        // don't encounter them.
+        // TODO: a more advanced way of doing things might be to ban trip
+        //        sequences to not find the same exact sequence, but also
+        //        individual trips that are too full.
+        tripsToBanInSubsequentSearches.push(tripId)
+
+        itineraryCapacity = Math.min(itineraryCapacity, remainingTripCapacity)
+      })
+
+      if (itineraryCapacity > 0) {
+        // itinerary is possible, add to list and update remaining group size.
+        // A field trip response is guaranteed to have only one itinerary, so it
+        // ok to set the itinerary by response as an array with a single
+        // itinerary.
+        assignedItinerariesByResponse[responseIdx] = [{
+          ...itinerary,
+          fieldTripGroupSize: Math.min(itineraryCapacity, remainingGroupSize)
+        }]
+        remainingGroupSize -= itineraryCapacity
+      }
+    })
+  })
+
+  return {
+    assignedItinerariesByResponse,
+    remainingGroupSize,
+    tripsToBanInSubsequentSearches
   }
 }
 
+/**
+ * Formats a field trip's travel date into the given format.
+ */
+function getFormattedRequestTravelDate (
+  request,
+  dateFormat = OTP_API_DATE_FORMAT
+) {
+  return moment(request.travelDate).format(dateFormat)
+}
+
+// These can be overridden in the field trip module config.
+const defaultFieldTripModeCapacities = {
+  'TRAM': 80,
+  'SUBWAY': 120,
+  'RAIL': 80,
+  'BUS': 40,
+  'FERRY': 100,
+  'CABLE_CAR': 20,
+  'GONDOLA': 15,
+  'FUNICULAR': 20
+}
+const unknownModeCapacity = 15
+
+/**
+ * Calculates the mode capacity based on the field trip module mode capacities
+ * (if it exists) or from the above default lookup of mode capacities or if
+ * given an unknown mode, then the unknownModeCapacity is returned.
+ *
+ * @param {Object} fieldTripModuleConfig the field trip module config
+ * @param {string} mode the OTP mode
+ */
+function getFieldTripGroupCapacityForMode (fieldTripModuleConfig, mode) {
+  const configModeCapacities = fieldTripModuleConfig?.modeCapacities
+  return (configModeCapacities && configModeCapacities[mode]) ||
+      defaultFieldTripModeCapacities[mode] ||
+      unknownModeCapacity
+}
+
+/**
+ * Dispatches the appropriate cleanup actions after making requests to find the
+ * itineraries for an inbound or outbound field trip journey.
+ */
+function doFieldTripPlanRequestCleanup (dispatch, searchId) {
```

Comment:
I don't see too much value in passing dispatch as a param vs. just using the boilerplate thunk syntax.

Reply:
Refactored into a thunk in a4480936984a2a1cb2d9dd6f97adc4ecd3a9a12c